src/transport/Session.cpp

```diff
@@ -46,5 +46,22 @@ OutgoingGroupSession * Session::AsOutgoingGroupSession()
     return static_cast<OutgoingGroupSession *>(this);
 }
 
+
+void Session::TryShiftToSession(const SessionHandle & session)
```

Comment:
There must be cross-checks that prevent the following:

- Shift sessions where peers mismatch
- Shift sessions across fabrics
- Shift sessions where CATs mismatch

Reply:
This is a private API (actually protected, because it is called from SecureSession), all checks are performed by the caller.